ci/check-worktree-is-clean

```diff
@@ -6,7 +6,7 @@ set -o nounset -o errexit -o pipefail
 # and this addresses that.
 git update-index -q --refresh
 
-if ! git diff-files --quiet; then
+if ! git diff-files --quiet -- . ':(exclude)**/pulumiManifest.go'; then
```

Comment:
Can we add a comment here with details about why?  This is certainly “weird” to be doing at this level - so some context on why will be helpful to future readers of this.